pkgs/applications/misc/dstask/default.nix

```
@@ -0,0 +1,31 @@
+{ stdenv, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "dstask";
+  version = "v0.18";
+  rev = "d687a0f9aec634a808cfa0df4aecb06ab4f57732";
+
+  src = fetchFromGitHub {
+    owner = "naggie";
+    repo = "dstask";
+    rev = "${version}";
+    sha256 = "16z5zlfj955pzsj0l58835slvpchdaq2vbyx2fjzi6y9xn1z2nd1";
+  };
```

```suggestion
  pname = "dstask";
  version = "0.18";

  src = fetchFromGitHub {
    owner = "naggie";
    repo = pname;
    rev = "v${version}";
    sha256 = "16z5zlfj955pzsj0l58835slvpchdaq2vbyx2fjzi6y9xn1z2nd1";
  };
```

`repo` is the same as `pname`

pkg versions shouldn't include "v"
(`buildFlagsArray` also needs updating)

Does `rev` in `buildGoModule` do anything? I can't find it there & I've not seen it used before.
https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/go-modules/generic/default.nix

---

Edit: ah, `rev` is for the `GIT_COMMIT` flag

---

Yes and for the command output: `dstask version`

---

Regarding the `repo = pname` thing: I actually think this is an antipattern. `repo` happens to be the same as `panme`, but the two things are not the same concept and don't necessarily change together.

Its pointless to have a nitpick war about this since we don't have an established standard. So no need to change anything. Just couldn't resist mentioning it ;)